packages/qwik-city/src/middleware/request-handler/user-response.ts

```diff
@@ -50,62 +52,88 @@ export function runQwikCity<T>(
     qwikSerializer,
     resolve!
   );
+
   return {
     response: responsePromise,
     requestEv,
     completion: asyncStore
-      ? asyncStore.run(requestEv, runNext, requestEv, resolve!)
-      : runNext(requestEv, resolve!),
+      ? asyncStore.run(requestEv, runNext, requestEv, rebuildRouteInfo, resolve!)
+      : runNext(requestEv, rebuildRouteInfo, resolve!),
   };
 }
 
-async function runNext(requestEv: RequestEventInternal, resolve: (value: any) => void) {
-  try {
-    // Run all middlewares
-    await requestEv.next();
-  } catch (e) {
-    if (e instanceof RedirectMessage) {
-      const stream = requestEv.getWritableStream();
-      await stream.close();
-    } else if (e instanceof ServerError) {
-      if (!requestEv.headersSent) {
-        const status = e.status as StatusCodes;
-        const accept = requestEv.request.headers.get('Accept');
-        if (accept && !accept.includes('text/html')) {
-          const qwikSerializer = requestEv[RequestEvQwikSerializer];
-          requestEv.headers.set('Content-Type', 'application/qwik-json');
-          requestEv.send(status, await qwikSerializer._serializeData(e.data, true));
-        } else {
-          const html = getErrorHtml(e.status, e.data);
-          requestEv.html(status, html);
+async function runNext(
+  requestEv: RequestEventInternal,
+  rebuildRouteInfo: RebuildRouteInfoInternal,
+  resolve: (value: any) => void
+) {
+  let rewriteAttempt = 1;
+
+  async function _runNext() {
+    try {
+      // Run all middlewares
+      await requestEv.next();
+    } catch (e) {
+      if (e instanceof RedirectMessage) {
+        const stream = requestEv.getWritableStream();
+        await stream.close();
+      } else if (e instanceof RewriteMessage) {
+        if (rewriteAttempt > 5) {
+          throw new Error(`Rewrite failed - Max rewrite attempts reached: ${rewriteAttempt - 1}`);
```

Comment:
Just put it at 50. 5 might happen after years of technical debt but 50 is just an infinite loop. Also, use save precious bytes, just say `Infinite rewrite loop`.

Reply:
@wmertens Done :)